### PR TITLE
fix: add csp frame ancestor header to mostly all responses

### PR DESCRIPTION
**Pull request by netroms**

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/security/utils/CspConstans.java [hunk 1]
```diff
@@ -30,9 +30,9 @@
 import java.util.List;
 import java.util.regex.Pattern;
 
-public class CspUtils
+public class CspConstans
```
Comment (amcgee) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/security/utils/CspConstans.java:33]: Typo - references will also need to be updated.

```suggestion
public class CspConstants
```

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/filter/CspFilter.java [hunk 2]
```diff
@@ -76,35 +78,33 @@ protected void doFilterInternal( HttpServletRequest req, HttpServletResponse res
...
         }
 
-        if ( isStaticContentInsideApi( req.getRequestURI() ) )
+        if ( LOGIN_PATTERN.matcher( url ).matches() )
         {
-            setFrameAncestorsCspRule( res );
-            chain.doFilter( req, res );
-            return;
-        }
+            String nounce = CodeGenerator.getRandomUrlToken();
```
Comment (amcgee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/filter/CspFilter.java:92]: Typo, references will also need to be updated

```suggestion
            String nonce = CodeGenerator.getRandomUrlToken();
```

File: dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/javascripts/login.js [hunk 3]
```diff
@@ -6,17 +6,7 @@ $( document ).ready( function()
...
-
-    $( '#2fa' ).click( function () {
-        $( '#2fa_code' ).attr("hidden", checked);
-        $( '#2fa_code' ).attr("readonly", checked);
-        document.getElementById( '2fa' ).checked = !checked;
-
-        checked = !checked;
-    });
-
-    $( '#loginForm').bind( 'submit', function()
```
Comment (amcgee) [inline dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/javascripts/login.js:19]: This change seems unrelated, and probably shouldn't be here?

**Review by amcgee — CHANGES_REQUESTED** [review]
Looks like there are several changes to login and account Struts pages that shouldn't be included in this PR, might have been accidentally included @netroms ?

File: dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/javascripts/jQuery/jquery-3.6.3.min.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/*! jQuery v3.6.3 | (c) OpenJS Foundation and other contributors | jquery.org/license */
```
Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/javascripts/jQuery/jquery-3.6.3.min.js:1]: Just checking, why is there a diff on the jquery lib file?

File: dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/useraccount/restore.vm [hunk 5]
```diff
@@ -16,44 +16,45 @@
...
+    <div id="accountInput">
 
-<h3>$i18n.getString( "restore_account" )</h3>
+        <h3>$i18n.getString( "restore_account" )</h3>
 
-<form id="restoreForm">
+        <form id="restoreForm">
 
-<input type="hidden" id="token" value="${token}">
+            <input type="hidden" id="token" value="${token}">
```
Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/useraccount/restore.vm:27]: Do we want these indentations?

**Review by larshelge — APPROVED** [review]

File: dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/javascripts/login.js (hunk 3 above)
Comment (netroms) [reply dhis-2/dhis-web/dhis-web-commons-resources/src/main/webapp/dhis-web-commons/javascripts/login.js:19]: I added back the CSP rules to the login page which was missing. Since it was related to the CSP filter. This method is just not in use anymore, so I cleaned it.

**Review by jbee — APPROVED** [review]

**Review by larshelge — APPROVED** [review]
